### PR TITLE
Platform dependent search bar and ui fixes

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
This adds two search bars one native for ios and the one for android is constructed in react-native code.

Also included fixes for the KeyboardAvoidView for select_server, login and mfa screens, as well as using platform specific styles to set the margin from the top on ios.

### CODE REVIEW

File: app/components/search_bar/search_bar.android.js [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    }
+
+    onCancelButtonPress() {
+        const onCancelButtonPress = this.props.onCancelButtonPress;
+        this.setState({
+            isOnFocus: false,
+            value: ''
+        });
+
+        if (onCancelButtonPress) {
```
Comment (jarredwitt) [inline app/components/search_bar/search_bar.android.js]: Thoughts on turning this into a defaultProp? You could do
```
defaultProps: {
    ...,
    onCancelButtonPress: () => true
}
```

That would eliminate the need for this logic because it's always going to be there whether it's passed in or not.

Comment (enahum) [inline app/components/search_bar/search_bar.android.js]: makes sense, I could do the same thing with the others...

**Comment by enahum** [comment]
@jarredwitt changes made

**Comment by lfbrock** [comment]
+1, updated the notice.txt slightly

File: app/scenes/mfa/mfa.js [hunk 2]
```diff
@@ -67,40 +76,48 @@ export default class Mfa extends Component {
...
-                    loading={false}
-                    containerStyle={GlobalStyles.signupButton}
-                >
-                    <FormattedText
-                        style={GlobalStyles.signupButtonText}
-                        id='mobile.components.select_server_view.proceed'
-                        defaultMessage='Proceed'
-                    />
-                </Button>
+                <TouchableWithoutFeedback onPress={this.blur}>
```
Comment (jarredwitt) [inline app/scenes/mfa/mfa.js]: Just wondering because I've never thought of using this, but why use TouchableWithoutFeedback instead of a ScrollView?

Comment (enahum) [inline app/scenes/mfa/mfa.js]: Two things scrollViews aren't very performant and the second thing is that I wanted to be able to tap the screen and execute the blur function so that way the keyboard gets hidden and there's no need to give other type of feedback to the user.

Comment (jarredwitt) [inline app/scenes/mfa/mfa.js]: Awesome, good to know.

File: app/components/search_bar/search_bar.android.js [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+    }
+
+    onBlur() {
+        const onBlur = this.props.onBlur;
+        this.setState({isOnFocus: false});
+        onBlur();
+        this.dismissKeyboard();
+    }
+
+    dismissKeyboard() {
```
Comment (jarredwitt) [inline app/components/search_bar/search_bar.android.js]: Minor nitpick, looks like this is only wrapping ```Keyboard.dismiss()```. Can we have ```onBlur``` and the ```onCancelButtonPress``` just call ```Keyboard.dismiss()``` directly since this method isn't doing anything other than calling ```Keyboard.dismiss()```? Or I guess we could leave it if there are plans to do other clean ups/actions when dismissing the keyboard.

Comment (enahum) [inline app/components/search_bar/search_bar.android.js]: will do

**Comment by enahum** [comment]
@jarredwitt done

File: app/components/channel_list/channel_list.js [hunk 4]
```diff
@@ -16,7 +16,11 @@ import deepEqual from 'deep-equal';
 const Styles = StyleSheet.create({
     container: {
         flex: 1,
-        marginTop: 20
+        ...Platform.select({
```
Comment (hmhealey) [inline app/components/channel_list/channel_list.js]: It can be a separate PR, but we might want to move this to the `RootLayout` so that it's applied to every scene automatically

File: app/components/search_bar/search_bar.android.js [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+        };
+
+        const inputStyle = {
+            paddingLeft: height * 0.5,
+            fontSize,
+            color: textColor
+        };
+
+        return (
+            <View style={{margin: 8}}>
```
Comment (hmhealey) [inline app/components/search_bar/search_bar.android.js]: Do we need two `View`s here or could the margin just be included in `styles.searchBar` or `searchBarStyle`?

File: app/scenes/channel/channel_header/channel_header.js [hunk 6]
```diff
@@ -24,8 +25,19 @@ export default class ChannelHeader extends React.PureComponent {
             theme
         } = this.props;
 
+        const containerStyle = {
+            backgroundColor: theme.sidebarHeaderBg,
+            flexDirection: 'row',
+            justifyContent: 'flex-start',
+            ...Platform.select({
```
Comment (hmhealey) [inline app/scenes/channel/channel_header/channel_header.js]: Same comment on moving this into the `RootLayout`

File: app/scenes/right_menu_drawer/right_menu_drawer.js [hunk 7]
```diff
@@ -18,7 +19,11 @@ const Styles = StyleSheet.create({
     container: {
         backgroundColor: '#2071a7',
         flex: 1,
-        paddingTop: 20
+        ...Platform.select({
```
Comment (hmhealey) [inline app/scenes/right_menu_drawer/right_menu_drawer.js]: Same comment on moving this into the `RootLayout`

File: app/components/search_bar/search_bar.android.js (hunk 5 above)
Comment (enahum) [inline app/components/search_bar/search_bar.android.js]: will do, I've missed it completely

File: app/components/channel_list/channel_list.js (hunk 4 above)
Comment (enahum) [inline app/components/channel_list/channel_list.js]: Wanna take that one and do it in a separate PR?

**Comment by enahum** [comment]
@hmhealey done

File: app/components/channel_list/channel_list.js (hunk 4 above)
Comment (hmhealey) [inline app/components/channel_list/channel_list.js]: Yep, will do
